src/ORM/DataExtension.php

```diff
@@ -13,7 +13,8 @@
 /**
  * An extension that adds additional functionality to a {@link DataObject}.
  *
- * @property DataObject $owner
+ * @template T of DataObject
+ * @extends Extension<T|static>
```

Comment:
What does the `|static` union do in this case?

Reply:
Ultimately T&static is the way to go, just tested it out and unfortunately phpstan resolves intersection types as *NEVER*. Would possibly need to include the open source silverstripe-phpstan to resolve